### PR TITLE
[PLAT-3917] Refactor automatic breadcrumbs into own space

### PR DESCRIPTION
**Pull request by nickdowell**
## Goal

The automatic breadcrumbs code was cluttering up the `BugsnagClient` class.

## Changeset

The bulk of NSNotification -> breadcrumb code now resides in `BSGNotificationBreadcrumbs`.

`BSGClient` still observes battery, orientation, and low memory notifications so that it can update its state metadata.

## Testing

New unit tests have been added to verify the creation of breadcrumbs from posted notifications.

### CODE REVIEW

File: Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+            NSLog(@"%@", name);
+            [_workspaceNotificationCenter addObserver:self
+                                             selector:@selector(addBreadcrumbForNotification:)
+                                                 name:name
+                                               object:nil];
+        }
+#endif
+        
+        // NSMenu events (Mac only)
+        for (NSNotificationName name in [self automaticBreadcrumbMenuItemEvents]) {
```
Comment (kattrali) [inline Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m:224]: should this `if TARGET_OS_OSX` conditional enclose the mac only bits in this loop?

File: Bugsnag/Client/BugsnagClient.m [hunk 2]
```diff
@@ -883,7 +796,7 @@ - (void)notifyOutOfMemoryEvent {
             // If the termination breadcrumb is set, the app entered a normal
             // termination flow but expired before the watchdog sentinel could
             // be updated. In this case, no report should be sent.
-            if ([name isEqualToString:kAppWillTerminate]) {
+            if ([name isEqualToString:@"App Will Terminate"]) {
```
Comment (kattrali) [inline Bugsnag/Client/BugsnagClient.m:799]: Maybe BSGNotificationBreadcrumbs.h should extern constants for this and the instance below?

**Review by kattrali — COMMENTED** [review]
Looks good, had a few small questions

File: Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m (hunk 1 above)
Comment (nickdowell) [reply Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m:224]: Probably, yes; much of this is the original copy and pasted logic!

Comment (nickdowell) [reply Bugsnag/Breadcrumbs/BSGNotificationBreadcrumbs.m]: Done 👍

File: Bugsnag/Client/BugsnagClient.m (hunk 2 above)
Comment (nickdowell) [reply Bugsnag/Client/BugsnagClient.m]: Done (in a slightly different way) 👍

**Review by kattrali — APPROVED** [review]

**Pull request merged**